src/waypoint.jsx

```diff
@@ -47,8 +52,8 @@ export default class Waypoint extends React.Component {
     if (!Waypoint.getWindow()) {
       return;
     }
-
-    this._handleScroll = this._handleScroll.bind(this);
+    this._handleScroll = this.props.throttleHandler(this._handleScroll)
+      .bind(this);
```

Comment:
I'll give this a try, but I think we might want this to be 
```
this.props.throttleHandler(this._handleScroll.bind(this));
```
instead. Notice the `.bind` being inside the parenthesis. 

This is because otherwise we can't use `this` inside `_handleScroll`, right? It would be good to have a test for this. Something like 
```jsx
 describe('with a throttleHandler that delays execution', () => {
      beforeEach(() => {
        this.props.throttleHandler = (scrollHandler) => setTimeout(scrollHandler, 1);
        scrollNodeTo(this.subject(), 100);
      });

      it('does not call the onEnter handler immediately', () => {
        expect(this.props.onEnter).not.toHaveBeenCalled();
      });
      
      it('calls the onEnter handler after the delay', (done) => {
        setTimeout(() => {
          expect(this.props.onEnter).toHaveBeenCalled();
          done();
        }, 2);
      });
    });
```